src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/commands/hmi/on_as_app_service_data_notification_from_hmi.cc

```diff
@@ -54,6 +56,39 @@ OnASAppServiceDataNotificationFromHMI::
 void OnASAppServiceDataNotificationFromHMI::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
   LOG4CXX_DEBUG(logger_, "Received an OnAppServiceData from HMI");
+
+  std::string service_id =
+      (*message_)[strings::msg_params][strings::service_data]
+                 [strings::service_id].asString();
+  AppService* service =
+      application_manager_.GetAppServiceManager().FindServiceByID(service_id);
+  if (!service) {
+    LOG4CXX_DEBUG(
+        logger_, "No published services exist with service ID: " << service_id);
+    return;
+  } else if (service->mobile_service) {
```

Comment:
@jacobkeeler  you have early return, you can avoid else condition. It would be a little but more easy to read with just another if.